### PR TITLE
stop awaiting derivedFiled on upload

### PR DESCRIPTION
**Pull request by JDuqn**

### CODE REVIEW

File: packages/ozone-api/ozone-api-upload/src/ozone-api-upload.ts [hunk 1]
```diff
@@ -350,16 +357,16 @@ export class UploadFileRequest implements XMLHttpRequestLike {
...
+			})
 	}
-	async _waitForTask(uploadEndResult: UploadFileId): Promise < string | undefined > {
+
+	_waitForTask(taskId: string): Promise<TaskResult | undefined> {
 		const taskClient = getDefaultClient().taskClient()
-		const taskHandler = taskClient.waitForTask<TaskResult>(uploadEndResult.uploadFileId)
-		const result = await taskHandler.waitResult
-		return result && result.mediaId
+		const taskHandler = taskClient.waitForTask<TaskResult>(taskId, { skipWaitingOnSubTask: true })
```
Comment (lmancel) [inline packages/ozone-api/ozone-api-upload/src/ozone-api-upload.ts:369]: Shouldn't that be an option to be passed to the module ? This lib is not only made to be used by flowr-admin, we should have the choice to wait for full completion.

**Review by lmancel — CHANGES_REQUESTED** [review]

File: packages/ozone-api/ozone-api-upload/src/ozone-api-upload.ts (hunk 1 above)
Comment (JDuqn) [reply packages/ozone-api/ozone-api-upload/src/ozone-api-upload.ts:369]: You're right. I'm making it an option to pass to the uploadFile function.
Do you know where it's used ? In the FlowR monorepo, I've only seen it used in flowr-admin.
The problem is that in the old implementation, uploadFile was returning a string (the media id) but in mine, it has to return both the mediaId and the subsequent tasks that are skipped.
I don't want to make two return types unless I'm forced to do so.

File: packages/ozone-api/ozone-api-upload/src/ozone-api-upload.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { OzoneConfig } from 'ozone-config'
 import { OzoneAPIRequest } from 'ozone-api-request'
 import { getDefaultClient } from 'ozone-default-client'
-import { UUID } from 'ozone-type'
+import { TaskHandlerOption } from '../../../ozone-typescript-client/src/taskClient/taskClient'
```
Comment (hubjac1) [inline packages/ozone-api/ozone-api-upload/src/ozone-api-upload.ts:4]: `import { TaskHandlerOption } from 'ozone-typescript-client/src/taskClient/taskClient'`

Comment (hubjac1) [reply packages/ozone-api/ozone-api-upload/src/ozone-api-upload.ts:4]: you may need to add it as a dependency

**Review by lmancel — APPROVED** [review]

File: packages/ozone-api/ozone-api-upload/src/ozone-api-upload.ts (hunk 1 above)
Comment (Azevedo-Dani) [reply packages/ozone-api/ozone-api-upload/src/ozone-api-upload.ts]: The ozone components can be used by others projects like Flowr-Connect ( phone application .. ) or before it was used by Adobe Plugin.

**Review by Azevedo-Dani — APPROVED** [review]

**Pull request merged**